test/task.e2e-spec.ts

```diff
@@ -0,0 +1,57 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { INestApplication } from '@nestjs/common';
+import * as request from 'supertest';
+import { TasksModule } from '../src/tasks/tasks.module';
+import { TasksService } from '../src/tasks/tasks.service';
+import { TasksRepository } from '../src/tasks/tasks.repository';
+import { AppModule } from '../src/app.module';
+
+describe('AppController (e2e)', () => {
+  let app: INestApplication;
+  const mockTasksRepository = {};
+
+  beforeEach(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [TasksModule, AppModule],
+      providers: [
+        TasksService,
+        {
+          provide: TasksRepository,
+          useValue: mockTasksRepository,
+        },
+      ],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
```

Comment:
` request = () => supertest(app.getHttpServer())`
add this line 
now you dont have to call app.getHttpServer() every time